scripts/upgrade

```diff
@@ -153,9 +148,39 @@ ynh_systemd_action --service_name=nginx --action=reload
 ynh_script_progression --message="Configuring MinIO client..." --weight=1
 
 pushd $mc_path
- 	ynh_exec_warn_less sudo -u $app ./mc alias set minio "https://$domain" "$admin" "$password" --api S3v4
+	ynh_exec_warn_less sudo -u $app ./mc --no-color alias set minio "https://$domain" "$admin" "$password" --api S3v4
 popd
 
+#=================================================
+# ENSURE DOWNWARD COMPATIBILITY
+#=================================================
+ynh_script_progression --message="Ensuring downward compatibility..."
+
+if ynh_compare_current_package_version --comparison lt --version 2022.12.12~ynh1
+then
+	ynh_script_progression --message="Migration from Filesystem to Single-Node Single-Drive mode"
+	pushd $datadir
+		for d in * ; do
+		ynh_script_progression --message="Migrating bucket $d"
+			mv "$d" "DATA_$d"
+			pushd $mc_path
+				ynh_exec_warn_less sudo -u $app ./mc mb minio/"$d"
+				ynh_exec_warn_less sudo -u $app ./mc mirror --preserve "$datadir/DATA_$d" minio/"$d"
+				# This is a hack, but it will make outline_ynh users' lifes much easier !
+				if [ "$d" == "outlinestorage" ]
+				then
+					ynh_exec_warn_less sudo -u $app ./mc policy set public minio/outlinestorage
+				else
+					ynh_script_progression --message="Bucket is migrated and objects are now mirrored. However, make sure to set properly access policy of bucket $d. That part of the migration can't be automated, sorry ! You can log into the minio console to perform your changes."
```

Comment:
Maybe worth to send an email to the user?

This helper can be useful for that: https://github.com/YunoHost-Apps/Experimental_helpers/blob/master/ynh_send_readme_to_admin/ynh_send_readme_to_admin__2

Reply:
Maybe, but I'm just trying not to spend  days on that app... I'll try to have a look.
Do you use, or know people using MinIO outside of the outline_ynh case ?

Reply:
I understand :)

I don't know. I use it for replicating data from another site.